File: array/cpp/MinimumPlatforms.cpp
#include <bits/stdc++.h>
using namespace std;

int minimumPlatform(int arr[], int dep[], int n){
    sort(arr,arr+n);
    sort(dep,dep+n);
    int cur_plat = 1;
    int max_plat = 1;
    int i = 1;
    int j = 0;
    while(i<n){
        if (arr[i]>dep[j]){
            i++;
            j++;
        }else{
            cur_plat+=1;
            if (cur_plat>max_plat){
                max_plat = cur_plat;
            }
            i++;
        }
    }
    return max_plat;
}

int main(){
    int arr[] = {900, 940, 950, 1100, 1500, 1800};
    int dep[] = {910, 1200, 1120, 1130, 1900, 2000};

    int n = sizeof(arr)/sizeof(arr[0]);

    cout<< minimumPlatform(arr,dep,n) << endl;
}

File: array/java/MinimumPlatforms.java
import java.util.Arrays;

public class MinimumPlatforms {
    int minimumPlatforms(int arr[], int dep[]){
        Arrays.sort(arr);
        Arrays.sort(dep);
        
        int curPlat = 0;
        int maxPlat = 0;
        
        int i = 0;
        int j = 0;

        while(i<arr.length){
            if (arr[i] > dep[j]){
                j+=1;
                i+=1;
            }else{
                curPlat += 1;
                if (curPlat>maxPlat){
                    maxPlat = curPlat;
                }
                i+=1;
            }
        }
        return maxPlat;
    }

    public static void main(String[] args){
        MinimumPlatforms mp = new MinimumPlatforms();
        //int arr[] = new int[]{900, 940, 950, 1100, 1500, 1800};
        //int dep[] = new int[]{910, 1200, 1120, 1130, 1900, 2000};

        int arr[] = new int[]{800, 1100};
        int dep[] = new int[]{1200, 1500};

        System.out.println(mp.minimumPlatforms(arr, dep));
    }
}
